### ISSUE TITLE
Lowest screen brightness is too bright - meizu pro 5

### ISSUE THREAD

**Issue by dark-eye** [ubports/ubports-touch#213]
- Device: Meizu pro 5
- Channel: Stable
- Build:15.04

## Steps to reproduce

1. Go to bed with wife/partner
2. Uncheck automatic brightness adjustment
3. Select the lowest brightness
4. Open youtube and see crash curse.
5. The wife/partner should now complain about the brightness of your phone...

## Expected behavoir
The lowest brightnessshould from the indecator shouldindecator equal to the brightness that is set by : 

sudo su ; echo 2 > /sys/class/backlight/pwm-backlight.0/brightness

## Actual behavior
The  Lowest brightness is brighter the is provided be the driver.

**Comment by NeoTheThird** [comment]
Can't reproduce: I'm a nerd and don't have a partner. Could you maybe provide some logs from `~/.cache/upstart/spouse/log.txt`?

**Comment by NeoTheThird** [comment]
On a more serious note: I agree that the minimum brightness settings could be lower on all devices.

**Comment by Stefano0101** [comment]
Yes, it's too bright, needs optimising I guess.

**Comment by mihaelmilea** [comment]
In the dark, my automatic brightness adjustment doesn't even go to the lowest setting, it goes only the next position after the lowest. If I want the lowest setting I have to disable the automatic setting and adjust it manually. Device: Pro 5.

**Comment by RaviPrasadWA** [comment]
i too dont have a partner or the pro5 phone , OK could someone here send me the tools of the trade so i could try to work on this issue

**Comment by alexanderisme** [comment]
Maybe this can help, because I reported the bug already long time ago, when everything was still on Launchpad: 
https://bugs.launchpad.net/canonical-devices-system-image/+bug/1618391

**Comment by mymike00** [comment]
yep, it affects all the devices (I have a BQ E4.5), so, please, change the title

**Comment by mymike00** [comment]
@dark-eye there is no `/sys/class/backlight/pwm-backlight.0/brightness` in my nexus 5 but I have `/sys/class/backlight/lm3630/actual_brightness`.
If with the brightness slider I put the lowest value, then my file contains `1`, while if I put the maximum the file contains `255`
So idk how we can set a lower brightness... not from that file for sure

Given that, @NeoTheThird Idk how is this issue easy to fix, why did you put the good first issue label?

**Comment by dark-eye** [comment]
@mymike00 I think this is a device specific bug.
I also noticed you can get the same result with : 
```
sudo mirbacklight 1
```

**Comment by mymike00** [comment]
Well, maybe in each device it's different, but all of them are affected by this bug, even if in a different way

**Comment by mariogrip** [comment]
This is a first issue since it's only defined by a config file `/usr/share/repowerd/device-configs/config-{device}.qml` her is an example from fp2 https://github.com/ubports/android_device_fairphone_fp2/blob/ubp-5.1/ubuntu/config.xml And it's true that some devices can't be less bright, but thats a hardware issue we cannot fix. this was a bug reported for pro5

**Comment by dark-eye** [comment]
@mariogrip apologizes on the stupid  question but If I made a change that seems to have work on which repository should i upload it? (there's seems to no android_device_meizu_turbo :( )

**Comment by Flohack74** [comment]
Meizu devices and also BQ cannot be built by us due to the closed device tree, hence there are no repos you can put this. Thats a limitation that will be hard to fix. Normally device-dependent things will come via the device tree overlays we do. Is your fix related to the repowerd config file? We might be able to fix this in the rootfs, but no idea.

**Comment by dark-eye** [comment]
Hey @Flohack74 

yes its just the config... do I guess we can add it to the root sf and it would fix the issue, do you know which repo should I add add the file to?

**Comment by Flohack74** [comment]
I dont know which repo ships this file, this will take some investigations. Also, it might have side effects on other devices, so this will mean good testing before we can accept this. Also, the config should contain a minimal set of options that is needed to perform this change

**Comment by mateosalta** [comment]
I think this is should be set to not the lowest, but a "safe low" that is still visible under normal light/medium - if you set lower manually the system has a safe guard that will reset it higher on lock/unlock, so you can never make it where you can not see what you are interacting with. There are situations where I would like lower (dark room with night clock, or completely off) but I think we should keep things at the default safe low - Perhaps a app could instead enable a quick switch for extreme low light situations.

**Comment by mymike00** [comment]
If we implement a safe guard, then we can make it dependent on the light sensor. when unlocking read the light sensor and calculate the safe guard level. then if the brightness set is under the safe guard, set the brightness to the safe guard

**Comment by Flohack74** [comment]
All that can be set up device-dependent in the repowerd config file. There is a mapping from candela units to the device specific values. Look here: https://github.com/Flohack74/android_device_samsung_s3ve3g/blob/ebe3014977ea3a881cd92c8826b07dd6892189b8/ubuntu/config-default.xml#L177 - in theory it must be possible to put higher numbers there where the array starts to make the minimum brighter

**Comment by mateosalta** [comment]
@mymike00 the safe guard is already there, if you set via mirbacklight lower it will automatically move the brightness back up after screen lock - also screen dimming has no effect if your brightness is already at lowest

**Comment by mymike00** [comment]
@mateosalta Oh, I misread your post. good!
